chain/network/src/routing/routing_table_actor.rs

```diff
@@ -306,63 +306,56 @@ impl RoutingTableActor {
         prune_edges_not_reachable_for: Duration,
     ) -> Vec<Edge> {
         let now = Instant::now();
-        let mut oldest_time = now;
+        // Save nodes on disk and remove from memory only if elapsed time from oldest peer
+        // is greater than `SAVE_PEERS_MAX_TIME`
+        if !(force_pruning
+            || (self.peer_last_time_reachable.values())
+                .any(|&last_time| now.saturating_duration_since(last_time) >= SAVE_PEERS_MAX_TIME))
+        {
+            return Vec::new();
+        }
 
         // We compute routing graph every one second; we mark every node that was reachable during that time.
         // All nodes not reachable for at last 1 hour(SAVE_PEERS_AFTER_TIME) will be moved to disk.
-        let peers_to_remove = self
-            .peer_last_time_reachable
-            .iter()
-            .filter_map(|(peer_id, last_time)| {
-                oldest_time = std::cmp::min(oldest_time, *last_time);
-                if now.saturating_duration_since(*last_time) >= prune_edges_not_reachable_for {
-                    Some(peer_id.clone())
-                } else {
-                    None
-                }
+        // TODO - use drain_filter once it becomes stable - #59618
+        let peers_to_remove = (self.peer_last_time_reachable.iter())
```

Comment:
@matklad I think it's better to write it with brackets to prevent `rustfmt` from splitting it in muiltiple lines.

I see a lot of code like this:
```rust
   &genesis
            .config
            .validators
            .iter()
            .map(|account_info| {
                (
                    account_info.account_id.clone(),
                    account_info.public_key.clone(),
                    account_info.amount,
                )
            })
            .collect::<Vec<_>>(),
```

This would be more readable to write it simply as
```rust
   &(genesis.config.validators.iter())
            .map(|account_info| {
                (
                    account_info.account_id.clone(),
                    account_info.public_key.clone(),
                    account_info.amount,
                )
            })
            .collect::<Vec<_>>(),
```

As a good practice, I think putty brackets around `(... .iter()) ` or  `(... .values())` makes code easier to read.  I really don't see any reason as to why you would want to split into a new line every time a dot appears.

A good rule of thumb, would be to split lines, such that, there is at most one function call per line. 
The following code, does make sense following that rule:
```rust
           (self.peer_last_time_reachable.values())
            .any(|&last_time| now.saturating_duration_since(last_time) >= SAVE_PEERS_MAX_TIME);
```

I probably should write this as a suggestion to `rustfmt`.

There is a reason, why we don't always split every time after every dot, some code would look ridiculous, for example:
```rust
   &genesis
            .config
            .validators
            .iter()
            .map(|account_info| {
                (
                    account_info
                        .account_id
                        .clone(),
                    account_info
                         .public_key
                          .clone(),
                    account_info
                          .amount,
                )
            })
            .collect::<Vec<_>>(),
```

Reply:
Ah, I see, that's a good point. Won't clippy complain about the extra parens though?

Reply:
@matklad No, `clippy` doesn't complain about it.